File: frontend/src/models/InterdictionPrecedentId.js
// models/InterdictionPrecedentId.js
class InterdictionPrecedentId {
    constructor(idShift, idShift1) {
        this.idShift = idShift;
        this.idShift1 = idShift1;
    }

    equals(other) {
        return other instanceof InterdictionPrecedentId && 
               this.idShift === other.idShift &&
               this.idShift1 === other.idShift1;
    }

    hashCode() {
        return `${this.idShift}-${this.idShift1}`;
    }
}

export default InterdictionPrecedentId;

File: backend/src/main/java/com/example/HopitalPlanningProject/model/InterdictionPrecedentId.java
package com.example.HopitalPlanningProject.model;

import jakarta.persistence.Embeddable;
import lombok.*;

import java.io.Serializable;
import java.util.Objects;

@Embeddable
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class InterdictionPrecedentId implements Serializable {
    private int idShift;
    private int idShift1;

    // Implémentation de equals et hashCode
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        InterdictionPrecedentId that = (InterdictionPrecedentId) o;
        return idShift == that.idShift && idShift1 == that.idShift1;
    }

    @Override
    public int hashCode() {
        return Objects.hash(idShift, idShift1);
    }
}